### ISSUE TITLE
Separate and Abstract Client vs Processor

### ISSUE THREAD

**Issue by Demetrioz** [Demetrioz/FarmCraft.Core#1]
Currently, the AzureServiceBus implementation of IMessageConsumer contains a single ServiceBusClient and ServiceBusProcessor. This is fine for only having a single connection and consumer in an application, but causes issues if you want to scale at all.

**Example**:
Notification Service that has multiple consumers to handle an increased load on the queue. 

There's also an issue if you want to consume messages from multiple queues, as you need two different ServiceBusProcessor objects, but there's only a single static one. 

Try to figure out a way to have the concept of a "Client" that connects to the message broker, and a "Processor" that receives incoming messages.

**AzureServiceBus**:
- Create Single ServiceBusClient per service / application
- Allow the creation of one or more ServiceBusProcessor to handle the actual messages

```c#
client = new ServiceBusClient(connectionString);
processor = client.CreateProcessor();
```

**RabbitMQ**:
- Create a single channel object per service / application
- Allow the creation of one or more EventingBasicConsumers (or equivilent) to handle actual messages

```c#
using(channel = connection)
{
  channel.QueueDeclare()

  var consumer = new EventingBasicConsumer(channel)
}
```

**Comment by Demetrioz** [comment]
Created a MessageBusService class that should run as a singleton and is responsible for making the connection to an Azure Service Bus. From there, ServiceBusPublisher and ServiceBusConsumer receive a MessageBusService in the constructor, and are responsible for creating a single ServiceBusSender or ServiceBusProcessor.

If needed, an actor can create multiple publishers / consumers from the main MessageBusService, and is then responsible for disposing of those publishers / consumers upon exit

**Comment by Demetrioz** [comment]
Resolved with PR #3
